core/node/state_keeper/src/keeper.rs

```diff
@@ -564,8 +572,11 @@ impl ZkSyncStateKeeper {
         batch_executor: &mut dyn BatchExecutor<OwnedStorage>,
         updates_manager: &mut UpdatesManager,
         protocol_upgrade_tx: Option<ProtocolUpgradeTx>,
+        state_restored: bool,
         stop_receiver: &watch::Receiver<bool>,
     ) -> Result<(), Error> {
+        let mut is_last_block_sealed = state_restored;
```

Comment:
IIUC, this var is actually redundant because it corresponds to whether `updates_manager` has next block params; see the other line comment about making those `Option`al. So it can be replaced by the `has_next_block_params(&self) -> bool` getter in `UpdatesManager`.